src/communication/NetworkClient.ts

```diff
@@ -69,21 +68,37 @@ const throwApiError = run(
   },
   findProperty =>
     function throwApiError(cause: unknown) {
-      if (findProperty(cause, 'response') && cause.response != undefined) {
-        throw ApiError.createFromResponse(cause.response as AxiosResponse<any>);
-      }
       throw new ApiError(findProperty(cause, 'message') ? String(cause.message) : 'An unknown error has occurred');
     },
 );
 
+/**
+ * Checks whether an API error needs to be thrown based on the passed result.
+ */
+async function processFetchResponse(response: ResponseWithIdempotencyKey) {
+  // Request was successful, but no content was returned.
+  if (response.status == 204) return true;
+  // Request was successful and content was returned.
+  const body = await response.json();
+  if (Math.floor(response.status / 100) == 2) {
+    return body;
+  }
+  // Request was not successful, but the response body contains an error message.
+  if (body) {
+    throw ApiError.createFromResponse(body, response.idempotencyKey);
+  }
+  // Request was not successful.
+  throw new ApiError('An unknown error has occurred');
+}
+
 interface Data {}
 interface Context {}
 
 /**
- * This class is essentially a wrapper around axios. It simplifies communication with the Mollie API over the network.
+ * This class is essentially a wrapper around fetch. It simplifies communication with the Mollie API over the network.
  */
 export default class NetworkClient {
-  protected readonly axiosInstance: AxiosInstance;
+  protected readonly request: (pathname: string, options?: RequestInit) => Promise<ResponseWithIdempotencyKey>;
```

Comment:
Could we add one or two sentences to explain what this function does, specifically how it differs from (vanilla) `fetch`?